src/main/java/nwoolcan/model/utils/UnitOfMeasure.java

```diff
@@ -7,53 +7,147 @@
     /**
      * Grams.
      */
-    GRAM,
+    GRAM("g") {
+        @Override
+        public boolean validateValue(final Number value) {
+            return true;
+        }
+    },
     /**
      * Milliliters.
      */
-    MILLILITER,
+    MILLILITER("ml") {
+        @Override
+        public boolean validateValue(final Number value) {
+            return true;
+        }
+    },
     /**
      * Basic units.
      */
-    UNIT,
+    UNIT("no") {
+        @Override
+        public boolean validateValue(final Number value) {
+            return validateWhenInteger(value);
+        }
+    },
     /**
      * Percentage.
      */
-    PERCENTAGE,
+    PERCENTAGE("%") {
+        @Override
+        public boolean validateValue(final Number value) {
+            return value.doubleValue() >= 0 && value.doubleValue() <= 100;
+        }
+    },
     /**
      * Celsius degrees.
      */
-    CELSIUS_DEGREE,
+    CELSIUS_DEGREE("C") {
```

Comment:
```suggestion
    CELSIUS_DEGREE("°C") {
```

Reply:
Not sure whether it works

Reply:
I've tried, gives me warning

Reply:
which warning?

Reply:
Not unified string or smth, you can reproduce it locally